### ISSUE TITLE
DETCI/ACI calculation in MCSCF_TWO_STEP

### ISSUE THREAD

**Issue by huangm29** [evangelistalab/forte#272]
I am running a GASSCF-DSRG calculation and I realized that MCSCF combined with DETCI used a much smaller R and E convergence even it is set to 1e-8.  This is not the case in MCSCF combined with ACI. This makes a difference in MCSCF results where they should do the exact same thing.  Interestingly, the MCSCF-DSRG with DETCI (where it does not converge CI fully in the beginning) seems to lead to a lower minimum than MCSCF with ACI. 

Is this designed to be like this? Which one is preferred to be used?  

The input of my DE
`import forte
memory 50gb

molecule co{
 C     0.000000     0.000000     0.567636
 O     0.000000     0.000000    -0.567636
}

set basis cc-pvqz

set {
scf_type            pk
reference           rohf
e_convergence       8
d_convergence       8
}

set forte {
MINAO_BASIS cc-pvtz-minao
job_type            none
avas                true
avas_diagonalize    true
avas_sigma  0.98
subspace ["C(1s)","C(2s)","C(2p)","O(1s)","O(2s)","O(2p)"]
}

Ezero, wfn = energy('forte', return_wfn=True)

set forte {
    avas false
  avas_diagonalize false
  ms 0.0
  multiplicity 1
  root_sym 0
  sci_enforce_spin_complete true
  spin_project_full         true 
  job_type              mcscf_two_step
  active_space_solver   detci
  active_ref_type       gas
# one_cycle             true
  rotate_mos         [1,1,2]
  gas1               [1,0,0,0]
  gas2               [5,0,2,2]
  avg_state          [[0,1,1],[2,1,1],[3,1,1]]
  gas1max            [2,1,1]
  avg_weight         [[0],[1],[1]]
  casscf_active_frozen_orbital [0]
  casscf_ci_solver      detci
  e_convergence         8  # energy convergence of the FCI iterations
  r_convergence         8  # residual convergence of the FCI iterations
  casscf_e_convergence  8  # energy convergence of the MCSCF iterations
  casscf_g_convergence  6  # gradient convergence of the MCSCF iterations
  casscf_diis_start     0 
}
Egasscf, wfn_gasscf = energy('forte', return_wfn = True, ref_wfn=wfn)

set forte {
  job_type newdriver
  rotate_mos []
  correlation_solver sa-mrdsrg
  corr_level pt2
  calc_type sa
  transition_dipoles true
}
edsrg = energy('forte', ref_wfn = wfn_gasscf)`

The outputs are attached for both DETCI and ACI. 

[detci_mcscf.txt](https://github.com/evangelistalab/forte/files/7205491/detci_mcscf.txt)
[aci_mcscf.txt](https://github.com/evangelistalab/forte/files/7205493/aci_mcscf.txt)

**Comment by lcyyork** [comment]
Yes, it is a bug of using DETCI in MCSCF_2STEP. The MCSCF_2STEP is trying to read the previous DETCI results as starting guess for Davidson-Liu. I thought I turned it off in the master because the procedure is not yet working.

What I did is simple. The DL solver just use the wave functions from the previous orbital optimization step. This means that the DL guess size is equal to the number of roots.

I am uncertain why it is not working. Perhaps it is the phase of orbital or CI is not fixed. It could also be the guess size now becomes too small. A better approach is probably copying the wave functions from the previous step to the current set of guesses and then orthogonalizing the guesses against the old wave functions.
